api/graphql/mutations/topic.js

```diff
@@ -1,4 +1,34 @@
-export function subscribe (userId, topicId, communityId, isSubscribing) {
-  return TagFollow.subscribe(topicId, userId, communityId, isSubscribing)
-  .then(() => ({success: true}))
+import { sanitize } from 'hylo-utils/text'
+
+export async function topicMutationPermissionCheck (userId, communityId) {
+  const community = await Community.find(communityId)
+  if (!community) {
+    throw new Error('That community does not exist.')
+  }
+  if (!await GroupMembership.hasActiveMembership(userId, community)) {
+    throw new Error("You're not a member of that community.")
+  }
+}
+
+export async function createTopic (userId, topicName, communityId) {
+  await topicMutationPermissionCheck(userId, communityId)
+  const name = sanitize(topicName)
+  const invalidReason = Tag.validate(name)
+  if (invalidReason) {
+    throw new Error(invalidReason)
+  }
+
+  const topic = await Tag.findOrCreate(name)
```

Comment:
the variable naming makes me a little queasy. But I think the best change will be to rename Tag to Topic, which is obv outside of the scope of this ticket.
I was working on a branch that had that change, but have abandoned it as it was going down a wrong path.